dashboard/src/shared/Secret.ts

```diff
@@ -9,16 +6,22 @@ import {
 
 export default class Secret {
   public static resourcesClient = () => new KubeappsGrpcClient().getResourcesServiceClientImpl();
-  public static async get(cluster: string, namespace: string, name: string) {
-    const u = url.api.k8s.secret(cluster, namespace, name);
-    const { data } = await axiosWithAuth.get<ISecret>(u);
-    return data;
-  }
 
-  public static async list(cluster: string, namespace: string, fieldSelector?: string) {
-    const u = url.api.k8s.secrets(cluster, namespace, fieldSelector);
-    const { data } = await axiosWithAuth.get<IK8sList<ISecret, {}>>(u);
-    return data;
+  public static async getDockerConfigSecretNames(cluster: string, namespace: string) {
+    const result = await this.resourcesClient().GetSecretNames({
+      context: {
+        cluster,
+        namespace,
+      },
+    });
+
+    const secretNames = [];
+    for (const [name, type] of Object.entries(result.secretNames)) {
```

Comment:
Perhaps less wordy with a [for..in](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/for...in) statement but fine either way